presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java

```diff
@@ -0,0 +1,131 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.tests.statistics.StatisticsAssertion;
+import com.facebook.presto.tpch.ColumnNaming;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.tpch.TpchTable;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.tests.statistics.Metric.OUTPUT_ROW_COUNT;
+import static com.facebook.presto.tests.statistics.MetricComparisonStrategies.absoluteError;
+import static com.facebook.presto.tests.statistics.MetricComparisonStrategies.defaultTolerance;
+import static com.facebook.presto.tests.statistics.MetricComparisonStrategies.noError;
+import static com.facebook.presto.tests.statistics.MetricComparisonStrategies.relativeError;
+import static com.facebook.presto.tests.tpch.TpchQueryRunner.createQueryRunnerWithoutCatalogs;
+import static java.util.Collections.emptyMap;
+
+public class TestTpchDistributedStats
+{
+    private StatisticsAssertion statisticsAssertion;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        DistributedQueryRunner runner = createQueryRunnerWithoutCatalogs(emptyMap(), emptyMap());
+        runner.createCatalog(
+                "tpch",
+                "tpch",
+                ImmutableMap.of("tpch.column-naming", ColumnNaming.STANDARD.name()));
+        statisticsAssertion = new StatisticsAssertion(runner);
+    }
+
+    @AfterClass
+    public void tearDown()
+    {
+        statisticsAssertion.close();
+        statisticsAssertion = null;
+    }
+
+    @Test
+    public void testTableScanStats()
+    {
+        TpchTable.getTables()
+                .forEach(table -> statisticsAssertion.check("SELECT * FROM " + table.getTableName())
+                        //TODO use noError
+                        .estimate(OUTPUT_ROW_COUNT, defaultTolerance()));
+    }
+
+    @Test
+    public void testFilter()
+    {
+        statisticsAssertion.check("SELECT * FROM lineitem WHERE l_shipdate <= DATE '1998-12-01' - INTERVAL '90' DAY")
+                //TODO fix estimates
+                .estimate(OUTPUT_ROW_COUNT, relativeError(-.55, -.45));
+    }
+
+    @Test
+    public void testJoin()
+    {
+        statisticsAssertion.check("SELECT * FROM  part, partsupp WHERE p_partkey = ps_partkey")
+                .estimate(OUTPUT_ROW_COUNT, relativeError(.95, 1.05));
```

Comment:
At first it looks a bit like ±5%, but it seems to actually mean "expect estimate twice as much as actual value" (i.e. poor estimate). Am i reading this correctly? If so, mark it with `// TODO improve estimate` or something.